cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/writer/MessagingLineageWriter.java

```diff
@@ -81,7 +83,18 @@ private void publishLineage(ProgramRunId programRunId, DataAccessLineage lineage
   }
 
   private void publish(MetadataMessage message) {
-    StoreRequest request = StoreRequestBuilder.of(topic).addPayload(GSON.toJson(message)).build();
+    String messageJson = GSON.toJson(message);
+    int publishSizeLimit = cConf.getInt(Constants.Metadata.MESSAGING_PUBLISH_SIZE_LIMIT);
```

Comment:
Read it once in the construction and have the limit as a field.

Reply:
Done.